crates/rpc/rpc/src/eth/error.rs

```diff
@@ -101,6 +101,23 @@ pub enum EthApiError {
     InternalJsTracerError(String),
     #[error(transparent)]
     CallInputError(#[from] CallInputError),
+    /// Optimism related error
+    #[error(transparent)]
+    #[cfg(feature = "optimism")]
+    Optimism(#[from] OptimismEthApiError),
+}
+
+/// Eth Optimism Api Error
+#[derive(Debug, thiserror::Error)]
+pub enum OptimismEthApiError {
+    /// Wrapper around a [hyper::Error].
+    #[cfg(feature = "optimism")]
+    #[error(transparent)]
+    HyperError(#[from] hyper::Error),
+    /// Wrapper around an [http::Error].
+    #[cfg(feature = "optimism")]
+    #[error(transparent)]
+    HttpError(#[from] http::Error),
 }
```

Comment:
Hoist the `#[cfg(feature = "optimism")]` macro invocation above the enum definition instead of repeating on each variant.

Reply:
Addressed in https://github.com/anton-rs/op-reth/pull/154